service/controller/v1/arm_templates/main.json

```diff
@@ -423,7 +423,7 @@
           "vmssOsImageVersion": {"value": "[parameters('workerNodes')[0].osImage.version]"},
           "vmssVmCustomData": {"value": "[parameters('workerCloudConfigData')]"},
           "vmssLbBackendPoolId": {"value": "[reference('ingress_load_balancer_setup').outputs.backendPoolId.value]"},
-          "vmssVnetSubnetId": {"value": "[reference('virtual_network_setup').outputs.masterSubnetID.value]"}
+          "vmssVnetSubnetId": {"value": "[reference('virtual_network_setup').outputs.workerSubnetID.value]"}
```

Comment:
How did you notice this? Can we ensure this with tests or something?

Reply:
I just looked at the templates :) This works in both cases. The previous version was potentially less secure, as there were less isolation between masters and workers.

Only e2e test can cover this. It also would have to to use azure-sdk-for-go to check to take the private IP of machines and assert they are in correct network.

My common sense says it isn't worth it at all.